src/components/forms/formUtils.js

```diff
@@ -0,0 +1,108 @@
+import { FormFieldTypes, commonRequiredProps, commonOptionalProps } from './forms';
+import { isNil, isFunction, isArray, isEmpty, isString } from 'lodash/fp';
+import { isEmailAddress } from '../../libs/utils';
+
+export const validateFormProps = (props) => {
+  const type = (!isNil(props.type) ? props.type : FormFieldTypes.TEXT);
+
+  const requiredProps = type.requiredProps || [];
+  const optionalProps = type.optionalProps || [];
+
+  requiredProps.push(...commonRequiredProps);
+  optionalProps.push(...commonOptionalProps);
+  optionalProps.push(...requiredProps);
+
+  const propKeys = Object.keys(props);
+
+  propKeys.forEach((prop) => {
+    if (!optionalProps.includes(prop)) {
+      throw `unknown prop ${prop}`;
+    }
+  });
+
+  requiredProps.forEach((requiredProp) => {
+    if (!propKeys.includes(requiredProp)) {
+      throw `prop ${requiredProp} is required`;
+    }
+  });
+
+  if (isFunction(type.customPropValidation)) {
+    type.customPropValidation(props);
+  }
+};
+
+
+export const customSelectPropValidation = (props) => {
+  if (!isArray(props.selectOptions)) {
+    throw 'prop \'selectOptions\' must be an array';
+  }
+
+  if (isEmpty(props.selectOptions)) {
+    throw '\'selectOptions\' cannot be empty';
+  }
+
+  const isStringArr = isString(props.selectOptions[0]);
+
+  props.selectOptions.forEach((opt) => {
+    if (isStringArr) {
+      if (!isString(opt)) {
+        throw 'all values in \'selectOptions\' must be string typed';
+      }
+
+      return;
+    }
+
+    if (isNil(opt.displayText)) {
+      throw 'every value in \'selectOptions\' needs a \'displayText\' field';
+    }
+  });
+};
+
+export const customRadioPropValidation = (props) => {
+  if (!isArray(props.options)) {
+    throw '\'options\' prop must be an array.';
+  }
+
+  props.options.forEach((opt) => {
+    if (isNil(opt.name)) {
+      throw 'each option in \'options\' prop must have a \'name\' field';
+    }
+
+    if (isNil(opt.text)) {
+      throw 'each option in \'options\' prop must have a \'text\' field';
+    }
+
+    if (!isNil(opt.type) && !['boolean', 'string'].includes(opt.type)) {
+      throw `radio group option types can be 'boolean' or 'string', not: '${opt.type}'`;
+    }
+  });
+};
+
+export const updateSelectDefaultValue = ({ selectOptions, defaultValue }) => {
```

Comment:
Would recommend moving this function back into the forms config declaration, just to reduce the distance between these. The other config functions (for like defaultValue) are being defined there too, i dont know, it just keeps it consistent. I can understand if you want to keep it here tho!